lib/rex/proto/http/client_request.rb

```diff
@@ -147,6 +151,44 @@ def to_s(headers_only: false)
           pstr << (opts['encode_params'] ? set_encode_uri(v) : v)
         end
       end
+
+      if opts['vars_form_data']
+        unless opts['vars_form_data'].is_a?(::Array)
+          raise ::ArgumentError, "request_cgi: The provided `form_data` option is not valid. Expected: Array, Got: #{opts['form_data'].class}"
+        end
+
+        file_data = Rex::MIME::Message.new
+        # Initialize or reuse the previous form data boundary to ensure idempotency
+        opts['vars_form_data_boundary'] ||= file_data.bound
+        file_data.bound = opts['vars_form_data_boundary']
+
+        opts['vars_form_data'].each do |file_hash|
+          # The name of the HTTP form field
+          field_name = file_hash['name'].is_a?(::String) ? file_hash['name'] : nil
+
+          # Should we default to 'application/octet-stream', nil, or HttpClient's default of 'text/plain'?
+          # https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+          mime_type = file_hash.fetch('mime_type', 'text/plain')
+
+          # Default to HttpClient's default option of '8bit'
+          encoding = file_hash.fetch('encoding', '8bit')
+
+          file_contents = get_file_data(file_hash['data'])
+
+          filename = file_hash.key?('filename') ? file_hash['filename'] : get_filename(file_hash['data']) || field_name
+
+          content_disposition = 'form-data'
+          content_disposition << "; name=\"#{field_name}\"" if field_name
+          content_disposition << "; filename=\"#{::CGI.escape(filename)}\"" if filename
+
+          file_data.add_part(file_contents, mime_type, encoding, content_disposition)
+        end
+
+        pstr += file_data.to_s
+      end
+
+      ctype ||= "multipart/form-data; boundary=#{opts['vars_form_data_boundary']}" if opts['vars_form_data_boundary']
+      ctype ||= 'application/x-www-form-urlencoded' if opts['method'] == 'POST'
```

Comment:
So in theory only one or the other should be used here but it seems like there is a case here whereby one could use both headers? Should we add additional checks around this or make this into an `if....elsif` statement?

Reply:
Oh right forgot to take that into account. Marking this as resolved 😄